lib/collaborative_filtering.py

```diff
@@ -75,23 +76,25 @@ def als_step(self, latent_vectors, fixed_vecs, ratings, _lambda, type='user'):
         """
         if type == 'user':
             # Precompute
-            YTY = fixed_vecs.T.dot(fixed_vecs)
-            lambdaI = numpy.eye(YTY.shape[0]) * _lambda
-
+            lambdaI = numpy.eye(self.hyperparameters['n_factors']) * _lambda
             for u in range(latent_vectors.shape[0]):
+                confidence = self.build_confidence_matrix(u)
```

Comment:
Can we specify the type 'user' for function call for clarity? Even if it is the default value.